python/lsst/ts/standardscripts/auxtel/calibrations/power_off_atcalsys.py

```diff
@@ -0,0 +1,142 @@
+
+    @classmethod
+    def get_schema(cls):
+        return None
+
+    async def configure(self, config):
+        # This script does not require any configuration
+        pass
+
+    def set_metadata(self, metadata):
+        """Compute estimated duration."""
+
+        metadata.duration = self.timeout_lamp_cool_down
+
+    async def run(self):
+        """Run script."""
+        await self.assert_components_enabled()
+
+        await self.checkpoint("Turning lamp off")
+        await self.switch_lamp_off()
+
+        await self.checkpoint("Closing the shutter")
+        await self.white_light_source.cmd_closeShutter.start(
+            timeout=self.timeout_close_shutter
+        )
+
+        await self.checkpoint("Waiting for lamp to cool down")
+        await self.wait_for_lamp_to_cool_down()
+
+        await self.checkpoint("Stopping chiller")
+        await self.white_light_source.cmd_stopChiller.start(timeout=self.cmd_timeout)
+
+    async def switch_lamp_off(self):
+        """Switches white light source lamp off"""
+        await self.white_light_source.evt_lampState.flush()
+
+        await self.white_light_source.cmd_turnLampOff.start(
+            timeout=self.timeout_lamp_cool_down
+        )
+
+    async def wait_for_lamp_to_cool_down(self):
+        """Confirm the white lamp has switched OFF and has cooled down
+        to proceed with stopping the chiller.
+
+        Raises
+        ------
+        TimeOutError:
+            If the lamp doesn't cool down and fails to turn off
+            in self.timeout_lamp_cool_down.
+        """
+        lamp_state = await self.white_light_source.evt_lampState.aget(
+            timeout=self.timeout_lamp_cool_down
+        )
+        self.log.info(
+            f"Lamp state: {ATWhiteLight.LampBasicState(lamp_state.basicState)!r}."
+        )
+        while lamp_state.basicState != ATWhiteLight.LampBasicState.OFF:
+            try:
+                lamp_state = await self.white_light_source.evt_lampState.next(
+                    flush=False, timeout=self.timeout_lamp_cool_down
+                )
+                cool_down_wait_time = (
+                    lamp_state.cooldownEndTime - lamp_state.private_sndStamp
+                )
+                self.log.info(
+                    f"Lamp state: {ATWhiteLight.LampBasicState(lamp_state.basicState)!r}."
+                    f"Need to wait {cool_down_wait_time/60} min"
+                )
+                asyncio.sleep(60)
+            except asyncio.TimeoutError:
+                raise RuntimeError(
+                    f"White Light Lamp failed to turn off after {self.timeout_lamp_warm_up} s."
+                )
+
+    async def assert_components_enabled(self):
+        """Check if ATWhiteLight is ENABLED
+
+        Raises
+        ------
+        RunTimeError:
+            If ATWhiteLight is not ENABLED"""
+        for comp in [self.white_light_source]:
+            summary_state = await comp.evt_summaryState.aget()
```

Comment:
Make sure you add a timeout to `aget` or it can block here for ever if there's missing data.